sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1194,13 +1194,21 @@ case class AddMonths(startDate: Expression, numMonths: Expression)
 }
 
 /**
- * Returns number of months between dates date1 and date2.
+ * Returns number of months between dates `date1` and `date2`.
+ * If `date1` is later than `date2`, then the result is positive.
+ * If `date1` and `date2` are on the same day of month, or both
+ * are the last day of month, time of day will be ignored. Otherwise, the
+ * difference is calculated based on 31 days per month, and rounded to
+ * 8 digits unless roundOff=false.
  */
 // scalastyle:off line.size.limit
 @ExpressionDescription(
   usage = """
-    _FUNC_(timestamp1, timestamp2[, roundOff]) - Returns number of months between `timestamp1` and `timestamp2`.
-      The result is rounded to 8 decimal places by default. Set roundOff=false otherwise."""",
+    _FUNC_(date1, date2[, roundOff]) - If `date1` is later than `date2`, then the result
```

Comment:
Last comment. Let's just leave `timestamp1` and `timestamp2` as were in the `usage` specifically..

Reply:
I'm going to leave it as date1, date2 because that also lines up with the variable names in the function definition.

Reply:
Yea, I got that they are consistent now but actually I think timestamp is more correct since the actual expected types here are timestmaps, not dates. Let's just leave them as were alone. It had a long review iterations.

Reply:
👍🏾

Reply:
sorry if I wasn't clear. mind changing date1 and date2 to timestamp1 and timestamp2 here in `usage`?